include/moqt/publish_track_handler.h

```diff
@@ -0,0 +1,267 @@
+
+            kNoPreviousObject,
+            kObjectDataComplete,
+            kObjectContinuationDataNeeded,
+            kObjectDataIncomplete,              ///< PublishObject() was called when continuation data remains
+
+            /// Indicates that the published object data is too large based on the object header payload size plus
+            /// any/all data that was sent already.
+            kObjectDataTooLarge,
+
+            /// Previous object payload has not been completed and new object cannot start in per-group track mode
+            /// unless new group is used.
+            kPreviousObjectNotCompleteMustStartNewGroup,
+
+            /// Previous object payload has not been completed and new object cannot start in per-track track mode
+            /// without creating a new track. This requires to unpublish and to publish track again.
+            kPreviousObjectNotCompleteMustStartNewTrack,
+        };
+
+        /**
+         * @brief  Status codes for the publish track
+         */
+        enum class Status : uint8_t
+        {
+            kOK = 0,
+            kNotConnected,
+            kNotAnnounced,
+            kPendingAnnounceResponse,
+            kAnnounceNotAuthorized,
+            kNoSubscribers,
+            kSendingUnannounce,
+        };
+
+
+        // --------------------------------------------------------------------------
+        // Public API methods that normally should not be overridden
+        // --------------------------------------------------------------------------
+
+        /**
+         * @brief Publish track handler constructor
+         *
+         * @param full_track_name       Full track name struct
+         * @param track_mode            The track mode to operate using
+         * @param default_priority      Default priority for objects if not specified in ObjectHeaderss
+         * @param default_ttl           Default TTL for objects if not specified in ObjectHeaderss
+         */
+        PublishTrackHandler(const FullTrackName& full_track_name,
+                            TrackMode track_mode,
+                            uint8_t default_priority,
+                            uint32_t default_ttl)
+          : BaseTrackHandler(full_track_name)
+          , track_mode_(track_mode)
+          , def_priority_(default_priority)
+          , def_ttl_(default_ttl)
+        {
+        }
+
+        // --------------------------------------------------------------------------
+        // Public Virtual API callback event methods to be overridden
+        // --------------------------------------------------------------------------
+
+        /**
+         * @brief Notification of publish status
+         * @details Notification of publishing status, such as when it's ready to publish or not ready to publish
+         *
+         * @param status        Indicates the status of being able to publish
+         */
+        virtual void StatusChanged(PublishTrackHandler::Status status) = 0;
```

Comment:
rename to be called OnTrackStatusChanged ??

Reply:
I don't see adding On makes it redundant. I still feel callback having On... is more readable

Reply:
My comment is specifically about prefixing callbacks with the word `On`. My C# experience is leaking into this (what I'm about to describe is the C# event naming convention), but what I have always been used to is that the callback is named with a corresponding verb (with the timeline it occurs reflected in the tense) and that a function prefixed "On" (e.g `On<EventName>()`) normally *triggers* the corresponding callback, not the callback itself.

E.g example usage of both being:

```cpp
// Handler.
virtual void StatusChanged(PublishTrackHandler::Status status) = 0;

// Example Trigger (if exposed, not saying to add this).
void OnStatusChanged(PublishTrackHandler::Status status) {
    StatusChanged(status);
};
```

So I would vote leave it as it is.